### ISSUE TITLE
For numeric inputs, research the possibility of setting the inputmode attribute when allowing negative numbers

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#1249]
- In In `src/component-library/number-input/numberInputUtils.ts` we have a function, `getInputMode` which derives the value of the [`inputmode` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) to set on the `<input>` element within our `NumberInput` and `StringifiedNumberInput` components. This helps the user's device display the correct software keyboard when necessary (e.g. on a phone).

- For positive numbers, we set `inputmode` to `decimal` or `numeric` and users get a nice keyboard with big number buttons.

    But MDN says:

    > Devices may or may not show a minus key (-).

    Because of this caveat in the MDN docs, we leave `inputmode` as `text` when the input allows negative numbers. Are we being overly cautious here? Are there _actually_ devices that don't show a minus key? Which ones? Can we set `inputmode` to `decimal` or `numeric` anyway? Or will there be users that can't enter negative numbers?

**Comment by Jinxiao0302** [comment]
I can help with this feature~

**Comment by seancolsen** [comment]
Thanks @Jinxiao0302 I've assigned it to you. If you have any questions, feel free to ask here, or in [Matrix](https://wiki.mathesar.org/en/community#matrix)

**Comment by seancolsen** [comment]
Ok this is interesting. Thanks for the research, @Jinxiao0302. 

Turns out I have a Samsung tablet! I just whipped up some HTML and tried this out on my tablet.

- With `inputmode='decimal'`, there's no way to enter a minus sign.

- With `inputmode='numeric'`, there's one button which reads `.-`. Pressing it once inserts a dot. Pressing it twice quickly inserts a minus sign. Pressing a holding does nothing. This is terrible design. The user _can_ insert a minus sign, but it's not super clear how to do it. 

Argh!! One more reason I hate this tablet!

One of the above research sources recommends using an input for telephone numbers instead. I tried setting `inputmode='tel'`. With that, I _can_ enter a minus sign. But the user experience is still not great. I need to click to see an additional symbol keyboard. And (more importantly) the number keyboard is inverted for phones vs numeric keypads, which ended up feeling really confusing when thinking of entering a numerical value.

All of this leads me to the conclusion that we should keep the current behavior. We can re-evaluate if needed. Perhaps if our current behavior is ever problematic for users, Samsung will have improved their software keyboards at that point!
